hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/KafkaOffsetGen.java

```diff
@@ -270,6 +280,15 @@ public KafkaOffsetGen(TypedProperties props) {
     return checkpointOffsetReseter ? earliestOffsets : checkpointOffsets;
   }
 
+  private Boolean checkLastCheckpointType(Option<String> lastCheckpointStr) {
```

Comment:
should we name this "isValidCheckpointType" or something? also, can you add java docs as to what validation we are doing here.